UKHomeOffice/nmsw-ui
# NMSW-1291 & NMSW-1295 Display people counts and created/submitted by details

# Ticket

NMSW-1291 & NMSW-1295

----

## AC

Display crew count
Display passenger count
Display Created by date
Display who created the declaration

Once submitted, replace Created date with Submitted date (and replace "Created:" with "Submitted:"

GIVEN A voyage as in any state
WHEN the user views the voyage list
THEN I can see the number of crew and passengers

GIVEN A draft voyage has state "Draft" OR "presubmitted"
WHEN the user views the voyage list
THEN the creation date and the name of the user that created the voyage is displayed

GIVEN A voyage has state submitted
WHEN the user views the voyage list
THEN the submitted date and the name of the user that created the voyage is displayed (Currently this can only be the same person)

GIVEN A voyage has state precancelled, cancelled or failed
WHEN the user views the voyage list
THEN hide the "Created date" line

----

## To test

- run app
- sign in
- view declaration list
- > you should see crew/passenger counts for all voyages
- > you should see the created by section if the voyage is in draft
- > you should see the submitted by section if the voyage is submitted
- > you should see no created/submitted by if voyage is presubmitted, precancelled, cancelled or failed

----

## Notes

It was easier to do both in one PR as the changes were in the same section of code
